chain-signatures/node/src/storage/protocol_storage.rs

```diff
@@ -469,6 +454,7 @@ impl<A: ProtocolArtifact> ProtocolStorage<A> {
 
         match result {
             Ok(artifact) => {
+                self.used.write().await.insert(id);
```

Comment:
Shouldn't you check the result? 

If this returns false, the artifact was already marked as used by another task. I guess it shouldn't happen if the reservation logic works out. But to be locally consistent, wouldn't it be better to return None if insertion failed?